lib/spack/spack/test/compilers/basics.py

```diff
@@ -338,83 +337,83 @@ def test_default_flags():
 
 # Verify behavior of particular compiler definitions.
 def test_arm_flags():
-    supported_flag_test("openmp_flag", "-fopenmp", "arm@1.0")
-    supported_flag_test("cxx11_flag", "-std=c++11", "arm@1.0")
-    supported_flag_test("cxx14_flag", "-std=c++14", "arm@1.0")
-    supported_flag_test("cxx17_flag", "-std=c++1z", "arm@1.0")
-    supported_flag_test("c99_flag", "-std=c99", "arm@1.0")
-    supported_flag_test("c11_flag", "-std=c11", "arm@1.0")
-    supported_flag_test("cc_pic_flag", "-fPIC", "arm@1.0")
-    supported_flag_test("cxx_pic_flag", "-fPIC", "arm@1.0")
-    supported_flag_test("f77_pic_flag", "-fPIC", "arm@1.0")
-    supported_flag_test("fc_pic_flag", "-fPIC", "arm@1.0")
-    supported_flag_test("opt_flags", ["-O", "-O0", "-O1", "-O2", "-O3", "-Ofast"], "arm@1.0")
+    supported_flag_test("openmp_flag", "-fopenmp", "arm@=1.0")
+    supported_flag_test("cxx11_flag", "-std=c++11", "arm@=1.0")
+    supported_flag_test("cxx14_flag", "-std=c++14", "arm@=1.0")
+    supported_flag_test("cxx17_flag", "-std=c++1z", "arm@=1.0")
+    supported_flag_test("c99_flag", "-std=c99", "arm@=1.0")
+    supported_flag_test("c11_flag", "-std=c11", "arm@=1.0")
+    supported_flag_test("cc_pic_flag", "-fPIC", "arm@=1.0")
+    supported_flag_test("cxx_pic_flag", "-fPIC", "arm@=1.0")
+    supported_flag_test("f77_pic_flag", "-fPIC", "arm@=1.0")
+    supported_flag_test("fc_pic_flag", "-fPIC", "arm@=1.0")
+    supported_flag_test("opt_flags", ["-O", "-O0", "-O1", "-O2", "-O3", "-Ofast"], "arm@=1.0")
```

Comment:
Not 100% understanding why all these need to change.  Do they? Can't you just change supported_flag_test to use satisfies?

Reply:
Let me check the logic again. In some cases a compiler is looked up, in some cases a compiler is defined. If it's looked up in config, `@1.0` should be enough; if it's defined, `@=1.0` makes most sense.

Reply:
My only concern would be when people have to write compiler specs in `compilers.yaml`.  I think if they *have* to write `@=` there to define a compiler, it's not great -- for the same reason requiring every `version()` directive to use `"=4.0"` wouldn't be so great.

Reply:
My only concern would be when people have to write compiler specs in `compilers.yaml`.  I think if they *have* to write `@=` there to define a compiler, it's not great -- for the same reason requiring every `version()` directive to use `"=4.0"` wouldn't be so great.

Reply:
I've reverted the change here. I guess I started fixing tests before I modified the from_dict function